docs/bugs.json

```diff
@@ -1,4 +1,17 @@
 [
+    {
+        "uid": "SOL-2022-7",
+        "name": "StorageWriteRemovalBeforeConditionalTermination",
+        "summary": "Calling Yul functions that conditionally terminate the transaction using ``return(...)`` or ``stop()`` may result in incorrect removals of storage writes.",
+        "description": "A call to a Yul function that conditionally terminates the transaction did not correctly mark all prior changes to storage as 'used', which may result in storage writes to be incorrectly removed. This happens in cases in which it had been valid to remove the store, if the Yul function in question never actually terminated the transaction. Conditional termination of the transaction within the same Yul function is not affected. In Solidity with via-IR code generation, any state change before a function conditionally calling ``return(...)`` or ``stop()`` in inline assembly, may be incorrectly removed, whenever it had been valid to remove the change without the ``return(...)`` or ``stop()``. Since the bug only occurs on calls to Yul functions, inlining may prevent the bug in some cases.",
```

Comment:
Maybe we should add some context on how the step works? What do you think about adding this before your text?

```
Yul optimizer has a step that detects and removes storage operations which write values that get overwritten before they are ever read. The step tracks all storage writes performed within a Yul function definition or the top-level Yul block and marks each of them as *used*, *unused* or *undecided*. When the control flow jumps to a different function, *undecided* operations should be changed to *used* unless the function is known to always revert. After the analysis, all operations storing unused values can be safely removed without altering the behavior of the contract. For the purpose of the analysis, for each function call it has to be determined whether it can revert, successfully terminate the whole transaction or return back to the caller and continue. More than one of these can be true due to conditionals.
```

Reply:
Maybe that's rather for the blog post to explain in more detail? I'd keep the description here as concise as possible, while conveying which cases are affected as precisely as possible... but yeah, maybe you're right that it helps for the understanding of the situation here to explain it briefly as well...

Reply:
Well, it's just that I don't expect people using this list to be that familiar with internals so I think we should either add some necessary context to understand it or make it even simpler and redirect people to the blog post.

Or maybe it would be better if these descriptions focused on how to tell if you are affected by the bug and what bad can happen and leave the details completely to the blog post. That's the info I'd be interested in if I were browsing the list. I would not expect to be able to understand what's happening just from this.